### PR TITLE
docs: changes to create scope api

### PR DESCRIPTION
**Pull request by AdithyaSSup**
Added flow diagram and changes to spec in create scope api

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -669,15 +675,17 @@ paths:
...
                       status:
                         type: string
-                        enum: ['UNKNOWN_CLIENT_ID']
+                        enum: ['UNKNOWN_CLIENT_ID_ERROR']
+                  - type: object
+                    properties:
+                      status:
+                        type: string
+                        enum: ['UNKNOWN_SCOPE_ERROR']
+
```
Comment (rishabhpoddar) [inline api_spec.yaml:688]: Why does this PR have changes for delete scope for client API?

Also, I just realised, we are using UNKNOWN_SCOPE_ERROR with many meanings. In some APIs, it means that the scope doesn't exist in the global scopes table. Whereas here it means that the client did not have the scope.

We should not change the meaning of the error codes this way. Remove the UNKNOWN_SCOPE_ERROR from this API, and instead, add a boolean like `didScopeExist` in the status: OK response of the delete API

File: api_spec.yaml [hunk 2]
```diff
@@ -439,7 +439,7 @@ paths:
         - OAuth2 Recipe
       operationId: addOAuth2Scope
       description: |
-        Create a scope
+        Create a scope, [click here](https://app.code2flow.com/7PHBCpE5Ygjt) for flow diagram
       parameters:
```
Comment (rishabhpoddar) [inline api_spec.yaml:443]: in flow diagram:
- we do not allow creation of scopes with the prefix of `supertokens.`

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 2 above)
Comment (AdithyaSSup) [reply api_spec.yaml:443]: Changes made

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml]: I think some commits might have missed while the previous PR was merged. Now the issue is resolved.

Comment (AdithyaSSup) [reply api_spec.yaml:687]: Changes made on delete client scope api response

Comment (rishabhpoddar) [reply api_spec.yaml:687]: The flow diagram is not synced with these changes..

File: api_spec.yaml (hunk 2 above)
Comment (rishabhpoddar) [reply api_spec.yaml:443]: The checking of scopes (in this and other APIs) where we are comparing the values to openid or supertokens.* should happen AFTER the normalising (trimming and lowercasing) the input scope. So please make this change to all diagrams. List down the links of the changed diagram in a comment.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml:687]: Checked again.. Changes are available
https://app.code2flow.com/flowcharts/6478d1783f4c2482c82c7b5a

File: api_spec.yaml (hunk 2 above)
Comment (AdithyaSSup) [reply api_spec.yaml:443]: Change's made for the scopes:,

Please find the updated links below:
- Delete client scope api -> https://app.code2flow.com/VSVAl8IVRU1h
- Create oauth2 Scope Api. -> https://app.code2flow.com/7PHBCpE5Ygjt
- Update oauth2 Scope Api. -> https://app.code2flow.com/WDizMrUkln5l
- Delete oauth2 Scope api. -> https://app.code2flow.com/XEw1xg7jvfkD

**Pull request merged**